lib/Releases.php

```diff
@@ -1,82 +1,65 @@
 
-use Symfony\Component\Cache\Adapter\FilesystemAdapter;
-use Symfony\Contracts\Cache\ItemInterface;
+use Github\Client;
 
 class Releases
 {
-	public function __construct($owner='LMMS', $repo='lmms')
+	public function __construct(Client $client, string $owner, string $repo)
 	{
-		$this->cache = new FilesystemAdapter();
-		$this->client = new \Github\Client();
-		$this->client->addCache($this->cache, [
-			'default_ttl' => 1800
-		]);
-		$this->json = $this->client->api('repo')->releases()->all($owner, $repo);
+		$this->json = $client->repo()->releases()->all($owner, $repo);
 		usort($this->json, function ($a, $b) {
 			return version_compare($b['tag_name'], $a['tag_name']);
 		});
 	}
 
-	public function latestAssets($pattern, $stable = true)
+	public function latestStableAssets(): array
 	{
-		foreach ($this->json as $index => $release) {
+		return $this->latestAssets(true);
+	}
+
+	public function latestUnstableAssets(): array
+	{
+		return $this->latestAssets(false);
+	}
+
+	private function latestAssets(bool $stable = true): array
+	{
+		foreach ($this->json as $release) {
 			if ($release['draft']) {
 				continue;
 			}
 			if ($release['prerelease'] === $stable) {
 				continue;
 			}
 
-			$assets = [];
-			foreach($release['assets'] as $asset)
-			{
-				if (preg_match($pattern, $asset['name']))
-				{
-					$asset['release'] = $release;
-					$asset['osname'] = $this->osName($asset['name']);
-					$asset['release']['index'] = $index;
-					array_push($assets, $asset);
-				}
-			}
-			return $assets;
+			return $this->mapAssetsFromJson($release);
 		}
+		return [];
 	}
 
-	public function latestAsset($pattern, $stable = true)
-	{
-		$assets = $this->latestAssets($pattern, $stable);
-		return $assets ? array_pop($assets) : null;
-	}
-
-	public function latestWin32Asset($stable = true)
+	private function mapAssetsFromJson(array $json): array
 	{
-		return $this->latestAsset('/.*-win32\.exe/i', $stable);
-	}
-
-	public function latestWin64Asset($stable = true)
-	{
-		return $this->latestAsset('/.*-win64\.exe/i', $stable);
-	}
-
-	public function latestOSXAssets($stable = true)
-	{
-		return $this->latestAssets('/.*\.dmg/', $stable);
-	}
-
-	public function latestLinuxAssets($stable = true)
-	{
-		return $this->latestAssets('/.*\.AppImage/', $stable);
+		return array_map(function (array $asset) use ($json) {
+			return new Asset(
+				platform: self::platformFromAssetName($asset['name']),
+				platformName: self::platformNameFromAssetName($asset['name']),
+				releaseName: $json['name'],
+				downloadUrl: $asset['browser_download_url'],
+				description: $json['body'],
+				gitRef: $json['tag_name'],
+				date: $asset['created_at']
+			);
+		}, $json['assets']);
 	}
 
 	/*
 	 * Get "32-bit", "64-bit", etc based on Download URL
 	 */
-	private function get_arch($text) {
+	private static function getArchitectureFromAssetName(string $assetName): string {
 		$arch64 = array('amd64', 'win64', 'x86-64', 'x86_64', 'x64', '64-bit', '.dmg');
```

Comment:
I don't know how important the "64-bit" tag is these days, but I'm wondering if we should add `aarch64` as well as `arm64` to the array.

Reply:
That's probably a good idea, as it's something we ought to support eventually, but I don't think it's in scope here. It might also be better to distinguish them from what we currently call "64-bit", although I don't know what names would be most user friendly. "64-bit Intel" and "64-bit ARM", perhaps?

Reply:
When this "64-bit" notation was added, LMMS for "32-bit" was the norm.  All these years later, it can probably be reversed, and we only distinguish the Windows 32-bit builds as "32-bit", which would simplify the above logic and prevent archs (e.g.) RISCV 64-bit from needing another patch down the road.

Reply:
In regards to it being in scope, Mac M1/M2 builds are long overdue and they'll likely show up in CI before they'll show up in the official downloads.

Reply:
Added, resolving.